src/seb/registered_models/translate_e5_models.py

```diff
@@ -0,0 +1,98 @@
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import torch
+from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+
+import seb
+from seb import models
+from seb.registered_models.e5_models import E5Wrapper
+from seb.registries import models
+
+
+class TranslateE5Model(seb.Encoder):
+    def __init__(self, model_name: str) -> None:
+        self.model_name = model_name
+        self.mdl = E5Wrapper(model_name)
+        self.trans_model: M2M100ForConditionalGeneration = (  # type: ignore
+            M2M100ForConditionalGeneration.from_pretrained("facebook/m2m100_418M")
+        )
+        self.trans_tokenizer = M2M100Tokenizer.from_pretrained("facebook/m2m100_418M")
+
+    def translate(self, sentence: str, src_lang: str) -> str:
+        self.trans_tokenizer.src_lang = src_lang
+        encoded_sent = self.trans_tokenizer(sentence, return_tensors="pt")
+        gen_tokens = self.trans_model.generate(
+            **encoded_sent, forced_bos_token_id=self.trans_tokenizer.get_lang_id("en")
+        )
+        return self.trans_tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)[
+            0
+        ]
+
+    def encode(
+        self,
+        sentences: list[str],
+        *,
+        task: seb.Task,  # noqa: ARG002
+        batch_size: int = 32,
+        **kwargs: Any,
+    ) -> torch.Tensor:
+        try:
+            src_lang = task.languages[0]
+        except IndexError:
+            # Danish is the default fallback if no language is specified for the task.
+            src_lang = "da"
```

Comment:
Is it better to select random between da/nb/sv? It is only really the identification task which is the problem. Actually, I would probably just forgo translation in that case (as it make no sense to translate it all to english).

Reply:
The problem then becomes that if we have a bitext mining task, it also has multiple languages, and then we would perform worse because we don't translate. I would keep it as is, I think language identification is a small price to pay for the rest. (I'm also not sure whether language identification is a good task anyway, I mean a good multilingual model should preferably only consider semantics, not language)

Reply:
Also, I think it's a bit hard to justify choosing randomly, and then we would get varying results depending on when we run the benchmark. I would also hope that since it is the same model doing all of the translation, it would still pick up on let's say Swedish, even if the specified source language is Danish. Unless we do language identification I would say it's best as is.

Reply:
> (I'm also not sure whether language identification is a good task anyway, I mean a good multilingual model should preferably only consider semantics, not language)

@Muennighoff and I discussed it briefly. I would say that a model should both encode semantics and the language used (the language used is also semantic I would argue, "long live the queen" is different from "længe leve dronningen". Similarly, you could imagine sentences regarding gun usage could be read quite differently depending on whether it was American English or Scandinavian)

Good models seem to be able to do both (though there is a trade-off) 

@Muennighoff what are your thoughts here?

Reply:
> Also, I think it's a bit hard to justify choosing randomly, and then we would get varying results depending on when we run the benchmark

Agree. I don't want that either. 

I think it is fine to roll with either Danish or embed naively without translation.

Reply:
Good points raised. I think this is a very big box to open for now, but maybe language identification is not the best task use here, and we should instead aim for some way to benchmark knowledge about language's and semantics relations.
For instance "I'm gonna go walk the dog" and "Jeg skal lige gå en tur med hunden" should preferably be language-agnostic (in my views). While the example with the queen should not. I'm thinking of a task where "Dronningen" is closer to "Margrethe II" while "queen" is closer to "Elizabeth II", but this is way beyond the scope of the pull request :D

 It seems tough, that we cannot avoid cutting corners with this translation-embedding pipeline, and if so, then I would cut one at language identification rather than at bitext mining. I would keep Danish the default, but if you think there's a better way to do it, we can discuss further.

Reply:
keep danish as the default. I believe this cuts to some of the issues of translate then embed pipelines.

Reply:
alrighty

Reply:
> a model should both encode semantics and the language used

Yeah I very much agree with that! I think in the long-term, we want to be abe to tell the model in the instruction whether it should produce a language-agnostic embedding or not